### PR TITLE
#SB-24944 fix: CON - Automated Unit test cases of sunbird-video-player

### PR DESCRIPTION
**Pull request by santoshilimi**

### CODE REVIEW

File: projects/sunbird-video-player/src/lib/services/util.service.spec.ts [hunk 1]
```diff
@@ -5,8 +5,18 @@ import { UtilService } from './util.service';
 describe('UtilService', () => {
   beforeEach(() => TestBed.configureTestingModule({}));
 
-  it('should be created', () => {
-    const service: UtilService = TestBed.get(UtilService);
+  it('should be created UtilService', () => {
+    const service = TestBed.inject(UtilService);
     expect(service).toBeTruthy();
   });
+  it('should call getTimeSpentText', () => {
```
Comment (HarishGangula) [inline projects/sunbird-video-player/src/lib/services/util.service.spec.ts:12]: check this

File: projects/sunbird-video-player/src/lib/services/util.service.spec.ts [hunk 2]
```diff
@@ -5,8 +5,18 @@ import { UtilService } from './util.service';
...
+    const service = TestBed.inject(UtilService);
+    const timeFormat = service.getTimeSpentText(61);
+    expect(timeFormat).toEqual('0:00');
+  });
+  it('should call uniqueId', () => {
+    const service = TestBed.inject(UtilService);
+    const timeFormat = service.uniqueId(6);
+    expect(timeFormat).not.toEqual('qIocVf');
```
Comment (HarishGangula) [inline projects/sunbird-video-player/src/lib/services/util.service.spec.ts:20]: not equal but it should be a string

File: projects/sunbird-video-player/src/lib/services/viewer.service.spec.ts [hunk 3]
```diff
@@ -1,14 +1,78 @@
...
+    service.preFetchContent();
+    expect(service.getQuestionSet).toHaveBeenCalledWith('1234');
+  });
+  it('should call raiseEndEvent', () => {
+    const service = TestBed.inject(ViewerService);
+    service.isEndEventRaised = true;
+    spyOn(service, 'calculateScore').and.callThrough();
+    spyOn(service['utilService'], 'getTimeSpentText').and.callFake(() => 'true');
+    service.raiseEndEvent();
+    expect(service['utilService'].getTimeSpentText).not.toHaveBeenCalledWith(10);
```
Comment (HarishGangula) [inline projects/sunbird-video-player/src/lib/services/viewer.service.spec.ts:75]: Check this value **toHaveBeenCalledWith(10);**

**Review by HarishGangula — APPROVED** [review]

File: projects/sunbird-video-player/src/lib/services/viewer.service.spec.ts (hunk 3 above)
Comment (santoshilimi) [reply projects/sunbird-video-player/src/lib/services/viewer.service.spec.ts:75]: Now mocking **visitedLength**  value  and passing that value to getTimeSpentText(visitedLength)

**Review by HarishGangula — APPROVED** [review]

**Pull request merged**